pandas/io/parquet.py

```diff
@@ -266,11 +278,18 @@ def to_parquet(df, path, engine='auto', compression='snappy', index=None,
         engine's default behavior will be used.
 
         .. versionadded 0.24.0
+    partition_cols : list, optional
+            Column names by which to partition the dataset
+            Columns are partitioned in the order they are given
+            The behaviour applies only to pyarrow >= 0.7.0 and fastparquet
+            For other versions, this argument will be ignored.
```

Comment:
Is it ignored, or doesn't it pass through to `write_table` and raise?

I would prefer that we raise for old pyarrows, so that a user doesn't think this is working when it isn't.

Also, I thought that fast parquet called it `partition_on`.

Reply:
Yes, we will be passing this key as partition_on